Source/Tools/LinuxEmulation/LinuxSyscalls/SyscallsSMCTracking.cpp

```diff
@@ -170,6 +170,17 @@ void SyscallHandler::InvalidateGuestCodeRange(FEXCore::Core::InternalThreadState
   InvalidateCodeRangeIfNecessary(Thread, Start, Length);
 }
 
+void SyscallHandler::MarkGuestCodeRoot(FEXCore::Core::InternalThreadState& Thread, uint64_t Address) {
+  if (!CodeMapWriter) {
+    return;
+  }
+
+  auto Region = LookupExecutableFileSection(Thread, Address);
+  if (Region && CodeMapWriter->IsWriteEnabled(*Region)) {
```

Comment:
LookupExecutableFileSection needs to fail if `!EntryIt->second.Resource->MappedFile`, otherwise it will dereference a null pointer which then manifests in an observed crashes here.